src/Browscap/Data/Expander.php

```diff
@@ -205,6 +211,8 @@ private function parseUserAgent(UserAgent $uaData, bool $liteUa, bool $standardU
         foreach ($uaData->getChildren() as $child) {
             $this->patternId['child'] = $i;
 
+            assert(is_array($child));
+
             if (isset($child['devices']) && is_array($child['devices'])) {
                 // Replace our device array with a single device property with our #DEVICE# token replaced
                 foreach ($child['devices'] as $deviceMatch => $deviceName) {
```

Comment:
fwiw, if we went to PHP 8+, the line L213 would need updating, since $deviceMatch is thought to be an `integer`:

```
PHP Fatal error:  Uncaught TypeError: str_replace(): Argument #2 ($replace) must be of type array|string, int given in /home/james/workspace/browscap-site/vendor/browscap/browscap/src/Browscap/Data/Expander.php:213
```

Reply:
> fwiw, if we went to PHP 8+, the line L213 would need updating, since $deviceMatch is thought to be an `integer`

`$deviceMatch` is a string

Reply:
How did you get this error? `$deviceMatch` should be an string.

Reply:
I was trying `browscap-site` out with PHP 8.1 (which I'm running locally); I'm not sure why `$deviceMatch` is sometimes an `int`, I didn't manage to track it; casting to `(string) $deviceMatch` where it gets used in `str_replace` might be OK; but it would be interesting to find out why it's sometimes an `int` nonetheless, to see if the issue can be fixed further up :)

Reply:
We have a device with this replacement `"2014818": "Xiaomi 2014818"`. It looks like json_decode converts this string into an integer.

Reply:
Hah! Funnily enough, this was recently discussed on the PHP internals list :grin: https://externals.io/message/116735